src/components/CollectionLink/CollectionLink.tsx

```diff
@@ -14,13 +14,11 @@ export const CollectionLink = ({
   href,
   label,
   background,
+  tagName = "div",
 }: CollectionLinkProps) => {
   const hasBackground = !!background;
 
-  const innerClasses = clsx(
-    styles.inner,
-    hasBackground && styles.hasBackground
-  );
+  const CustomTag = tagName as keyof JSX.IntrinsicElements;
```

Comment:
Good job but typecasting/"using as" could cause issue if someone were to pass in an invalid string. Instead of typing `tagName` as `string` and casting to `keyof JSX.IntrinsicElements`, type it directly like so: https://stackoverflow.com/a/55970164. Only difference from that link would be to keep `tagName` optional.

Reply:
Also good to check out how they rename `tag` to `Wrapper` directly in the object spread